error.go

```diff
@@ -27,39 +27,45 @@ var _ fmt.Formatter = (*Error)(nil)
 // Dynamic properties is a brittle mechanism and should therefore be used with care and in a simple and robust manner.
 func (e *Error) WithProperty(key Property, value interface{}) *Error {
 	errorCopy := *e
+	errorCopy.properties = errorCopy.properties.with(key, value)
+	return &errorCopy
+}
 
-	if errorCopy.properties == nil {
-		errorCopy.properties = make(map[Property]interface{}, 1)
-	} else {
-		errorCopy.properties = make(map[Property]interface{}, len(e.properties) + 1)
-		for k, v := range e.properties {
-			errorCopy.properties[k] = v
-		}
+// WithNewProperty adds a dynamic property to error instance if it were not already set.
+// If an error already contained another value for the same property, it is NOT overwritten.
+func (e *Error) WithNewProperty(key Property, value interface{}) *Error {
```

Comment:
I'm not sure the name is good enough. 
And even so, if this method may not do anything, shouldn't it return bool or error on property conflict?
Could you please describe some real life example for this method?

Reply:
> And even so, if this method may not do anything, shouldn't it return bool or error on property conflict?

In my use case, it just prevents double write of same value, therefore, it doesn't need to signal anything. And I want to save liquid interface.

> I'm not sure the name is good enough.

Which name do you suggest?

Reply:
Ok, I decided to remove this method. In my use case, I will add private function.